src/arfsdao.ts

```diff
@@ -487,65 +524,33 @@ export class ArFSDAO extends ArFSDAOAnonymous {
 		metadataRewardSettings: RewardSettings,
 		destFileName?: string
 	): Promise<ArFSUploadPrivateFileResult> {
-		// Establish destination file name
-		const destinationFileName = destFileName ?? wrappedFile.getBaseFileName();
-
-		// Generate file ID
-		const fileId = uuidv4();
-
-		// Gather file information
-		const { fileSize, dataContentType, lastModifiedDateMS } = wrappedFile.gatherFileInfo();
-
-		// Read file data into memory
-		const fileData = wrappedFile.getFileDataBuffer();
-
-		// Build file data transaction
-		const fileDataPrototype = new ArFSPrivateFileDataPrototype(
-			await ArFSPrivateFileDataTransactionData.from(fileData, fileId, driveKey)
-		);
-		const dataTrx = await this.prepareArFSObjectTransaction(fileDataPrototype, fileDataRewardSettings);
-
-		// Upload file data
-		if (!this.dryRun) {
-			console.log(`Uploading private file: "${wrappedFile.filePath}" to the permaweb..`);
-			await this.sendChunkedUploadWithProgress(dataTrx);
-		}
-
-		// Prepare meta data transaction
-		const fileMetaData = await ArFSPrivateFileMetadataTransactionData.from(
-			destinationFileName,
-			fileSize,
-			lastModifiedDateMS,
-			dataTrx.id,
-			dataContentType,
-			fileId,
-			driveKey
-		);
-		const fileMetadataPrototype = new ArFSPrivateFileMetaDataPrototype(
-			fileMetaData,
-			driveId,
-			fileId,
-			parentFolderId
+		return this.uploadFile(
```

Comment:
Cool, this is inferring the generic types to pass through here. I noticed there there was no types here `uploadFile<myTypes>`